linkerd/app/outbound/src/lib.rs

```diff
@@ -68,9 +68,47 @@ impl Config {
     > + Unpin
            + Clone
            + Send {
+        let connect = svc::connect(self.proxy.connect.keepalive).into_inner();
         // Establishes connections to remote peers (for both TCP
         // forwarding and HTTP proxying).
-        svc::connect(self.proxy.connect.keepalive)
+        self.build_tcp_connect_with(connect, local_identity, metrics)
+    }
+
+    pub fn build_tcp_connect_with<C>(
+        &self,
+        connect: C,
+        local_identity: tls::Conditional<identity::Local>,
+        metrics: &ProxyMetrics,
+    ) -> impl tower::Service<
+        TcpEndpoint,
+        Error = Error,
+        Future = impl future::Future + Send,
+        Response = impl tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static,
+    > + tower::Service<
+        HttpEndpoint,
+        Error = Error,
+        Future = impl future::Future + Send,
+        Response = impl tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static,
+    > + Unpin
+           + Clone
+           + Send
+    where
+        C: tower::Service<HttpEndpoint, Error = std::io::Error> + Clone + Send + Unpin,
+        <C as tower::Service<HttpEndpoint>>::Future: Send + 'static,
+        <C as tower::Service<HttpEndpoint>>::Response:
+            tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static,
+        transport::BoxedIo: From<<C as tower::Service<HttpEndpoint>>::Response>
+            + From<tokio_rustls::client::TlsStream<<C as tower::Service<HttpEndpoint>>::Response>>,
+        C: tower::Service<TcpEndpoint, Error = std::io::Error> + Clone + Send + Unpin,
+        <C as tower::Service<TcpEndpoint>>::Future: Send + 'static,
+        <C as tower::Service<TcpEndpoint>>::Response:
+            tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send + 'static,
```

Comment:
If we do need to keep this, let's please use another intermediary type attr inttead of having to do all of the `as tower::Service` noise